PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/storage/AttachStorageEmulatorAccountAction.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.serviceexplorer.azure.storage;
+
+import com.intellij.openapi.project.Project;
+import com.microsoft.intellij.forms.ExternalStorageAccountForm;
+import com.microsoft.tooling.msservices.helpers.ExternalStorageHelper;
+import com.microsoft.tooling.msservices.helpers.Name;
+import com.microsoft.tooling.msservices.model.storage.ClientStorageAccount;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.storage.ExternalStorageNode;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.storage.StorageModule;
+
+import javax.swing.*;
+
+@Name("Attach storage emulator...")
+public class AttachStorageEmulatorAccountAction extends NodeActionListener {
+    private final String connectionName = "devstoreaccount1";
+    private final StorageModule storageModule;
+
+    public AttachStorageEmulatorAccountAction(StorageModule storageModule) {
+        this.storageModule = storageModule;
+    }
+
+    @Override
+    public void actionPerformed(NodeActionEvent e) {
```

Comment:
It might be I miss something, but don't we need to validate this action against platform? As far as I understood, this feature is available on Windows only. Also, from documentation - "The storage emulator uses a local Microsoft SQL Server instance". Should we check for local SQL Server instance to make it works?

Reply:
There is also a cross-platform one ([Azurite](https://github.com/Azure/Azurite))

Reply:
True, but it might be not set on client box. TBD: (Since storage emulator runs locally) do we need to check for all conditions to be sure that action can be executed?

Reply:
Tough one. Even on Windows, if it is not running it is possible to add the node but not to connect to it.